ephios/core/signup/views.py

```diff
@@ -53,12 +53,14 @@ def setup(self, request, *args, **kwargs):
     def form_valid(self, form):
         choice = form.cleaned_data["signup_choice"]
         validator = self.shift.signup_flow.get_validator(self.participant)
-        if choice == "sign_up" and validator.can_sign_up():
-            return self.signup_pressed(form)
-        if choice == "customize" and validator.can_customize_signup():
-            return self.customize_pressed(form)
-        if choice == "decline" and validator.can_decline():
-            return self.decline_pressed(form)
+        with transaction.atomic():
+            Shift.objects.select_for_update().get(pk=self.shift.pk)
```

Comment:
does this work inside the transaction? the example in the docs always applies the lock before the transaction

Reply:
Beware that in the example [in the docs](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#select-for-update) the queryset is evaluated inside the transaction, meaning that the lock is acquired inside the transaction, just like here where `.get()` evaluates the query.